.github/workflows/ci.yml

```diff
@@ -0,0 +1,52 @@
+name: CI
+
+on:
+  push:
+    branches: [master, test]
+  pull_request:
+    branches: [master, test]
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu, windows, macos]
+        python-version: ['3.5', '3.6', '3.7', '3.8', '3.9', 'pypy3']
+        include:
+          # https://github.com/pydoit/doit/issues/372
+          - os: macos
+            pytest-args: -k 'not(cmd_auto or TestFileWatcher)'
+          - os: macos
+            python-version: pypy3
+            pytest-args: -k 'not(cmd_auto or TestFileWatcher or remove_all or ForgetAll)'
+        exclude:
+          - os: windows
+            python-version: pypy3
+    steps:
+      - if: ${{ matrix.os == 'ubuntu' }}
+        run: sudo apt-get install strace
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - run: pip install --user -U pip wheel setuptools
+      - id: pip-cache
+        run: echo "::set-output name=dir::$(pip cache dir)"
+      - uses: actions/cache@v2
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: ${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('setup.py', 'dev_requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-${{ matrix.python-version }}-
+            ${{ runner.os }}-pip-
+      - run: pip install . -r dev_requirements.txt python-coveralls
+      - run: pip freeze
+      - run: pip check
+      - run: doit pyflakes
+      - run: py.test -vv ${{ matrix.pytest-args }}
+      - if: ${{ matrix.os == 'ubuntu' && matrix.python-version == '3.8' }}
+        run: |
+          doit coverage
+          coveralls || echo "TODO: remove when run on a repo that is set up"
```

Comment:
What this means? "which kind of repo setup"?

Reply:
I hadn't set up codecov in a while, since the bad old days when a token was needed. I assume it's just working now?

Reply:
Ok understood. No, it is not working:

```
INFO:coveralls:422
INFO:coveralls:{"message":"Couldn't find a repository matching this job.","error":true}
TODO: remove when run on a repo that is set up
```

I dont understand this error message. It seems to work well on travis... Any idea why ?

Reply:
Yeah, there's probably a token or something. And looking at it, of course, it's `coveralls`. It looks it does need a token in secrets, which I don't like to bother with until i absolutely must.

I have been using `codecov` recently, which doesn't require a secret for public repos, and only generates _one_ PR comment, which it updates, which I find more convenient... will do a follow-up PR with a demo.

Reply:
PR up https://github.com/pydoit/doit/pull/384 with lots of :heavy_check_mark: